### PR TITLE
HPCC-27211 process dynamic shared lib path for files under lib2 separately

### PR DESCRIPTION
**Pull request by xwang2713**
The root problem of that the files under lib2/ may also depend other libraries in this directory. But the process order may leave some libraries improbably handled. For example libicui18n.60.2.dylib depends on lib2/libicuuc.60.2.dylib. If libicui18n.60.2.dylib is processed before libicuuc.60.2.dylib adding to lib2 directory the dependency of libicuuc.60.2.dylib is not correctly updated.

The current fix is to process all files under lib2 after all of them added to lib2 directory.

Also fixed JIRA HPCC-27193  extra "endif()" in lib2/CMakeLists.txt
Signed-off-by: Xiaoming Wang <xiaoming.wang@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by xwang2713** [comment]
Private fix to verify: http://10.224.20.10/builds/custom/JIRA/HPCC-27211/hpccsystems-clienttools-community_8.6.5-closedown0Darwin-x86_64.dmg

File: lib2/CMakeLists.txt [hunk 1]
```diff
@@ -115,37 +147,33 @@ foreach(dylib ${DYLIBS})
...
-                     ERROR_VARIABLE  err
-                  )
-               endif()
+               ${update_dynamic_shared_lib_path}
            endforeach ()
         ")
    endif()
 endforeach(dylib)
+
+# Update dynmaic shared library path for libraries under lib2/
```
Comment (richardkchapman) [inline lib2/CMakeLists.txt:160]: typo: dynamic

**Review by richardkchapman — CHANGES_REQUESTED** [review]

**Comment by dcamper** [comment]
> @dcamper a private fix to verify: http://10.224.20.10/builds/custom/JIRA/HPCC-27211/hpccsystems-clienttools-community_8.6.5-closedown0Darwin-x86_64.dmg

I just tested this and it appears to work (using the simple tests found in the Jira):

[2022-02-22 07:15:06 -0600]
campda01@HBECAMPDA01-MAC:/opt/HPCCSystems/8.6.5/clienttools/bin $ ./ecl --version
community_8.6.5-closedown0[remotes/origin/HPCC-27211-dynamic-shared-lib-path-4-files-in-lib2-0-g4ee718]

[2022-02-22 07:15:10 -0600]
campda01@HBECAMPDA01-MAC:/opt/HPCCSystems/8.6.5/clienttools/bin $ ./eclcc --version
8.6.5 community_8.6.5-closedown0[remotes/origin/HPCC-27211-dynamic-shared-lib-path-4-files-in-lib2-0-g4ee718]

**Review by Michael-Gardner — APPROVED** [review]
Approving. I had noticed that the get_filename_component cmake call had changed in the latest cmake release, but had no difficulty building.

File: lib2/CMakeLists.txt (hunk 1 above)
Comment (xwang2713) [reply lib2/CMakeLists.txt]: Fixed and rebased

**Review by richardkchapman — APPROVED** [review]

**Pull request merged**